server/src/main/java/org/elasticsearch/action/bulk/BulkRequestParser.java

```diff
@@ -115,6 +118,7 @@ public void parse(
         int line = 0;
         int from = 0;
         byte marker = xContent.streamSeparator();
+        final Map<String, String> stringDeduplicator = new HashMap<>();
```

Comment:
If we are going to do this, can we at least add a comment here explaining the intent? Specifically noting that this doesn't save on the actual instantiation of the duplicate strings, only on the lifetime of those duplicates (can be GC'd after parsing is complete).

Reply:
Sure thing, done in 02e3fc42f7b